File: iOS/EvianBlog/api/projects/ProjectAPI.swift
//
//  ProjectAPI.swift
//  EvianBlog
//
//  Created by Evian张 on 2020/5/18.
//  Copyright © 2020 Evian张. All rights reserved.
//

import Alamofire

import Foundation
import Combine

extension BlogAPI {
	func getProjects() -> AnyPublisher<[Project], BlogAPIError> {
		self.fetch(url: Endpoint.Projects.getProjects)
	}
}

File: Android/app/src/main/java/top/evian_zhang/evianblog/api/projects/ProjectAPI.kt
package top.evian_zhang.evianblog.api.projects

import top.evian_zhang.evianblog.api.Project
import top.evian_zhang.evianblog.api.BlogAPI
import top.evian_zhang.evianblog.api.Endpoint

suspend fun BlogAPI.getProjects(): List<Project> {
    return this.fetch(Endpoint.Projects.getProjects)
}
